### DISCUSSION TITLE
Debian/Ubuntu apt repository?

### DISCUSSION THREAD

**Discussion by elsiehupp** [any1/wayvnc#93] [General]
I don't see any method for setting up WayVNC on Debian/Ubuntu other than [building it from source](https://github.com/any1/wayvnc/issues/81). What might be involved in making it available from `apt`?

**Comment by danshick** [comment]
https://www.debian.org/doc/manuals/distribute-deb/distribute-deb.html

https://wiki.ubuntu.com/UbuntuDevelopment/NewPackages

**Comment by crazy-logic** [comment]
@elsiehupp - have you managed to get it to compile under Ubuntu 21.04 at all? or has anyone made any progress with Gnome?

**Reply by any1** [reply]
I think Gnome has its own VNC solution.

**Reply by crazy-logic** [reply]
Is does but it's user session only so doesn't work with lock screens, multiple user sessions or first logon of a user. so it's screen sharing but not remote access.

**Reply by any1** [reply]
I see. Gnome support could be added, but it depends on whether there are ways to do screen capturing and virtual inputs.

**Reply by elsiehupp** [reply]
I nerfed my Ubuntu installation a few months back—thanks LVM!—and I haven’t gotten around to reinstalling it, so I just have Debian for now. I’ve been using the built-in VNC server on Debian, but it’s definitely annoying that it can’t handle login screens, which makes it quite a bit less useful if, say, you’ve just rebooted your computer.

The other issue I had been having with GNOME VNC is that the authentication it uses isn’t supported by the first-party macOS VNC client, but the solution I came up with for that was just to spend $$$ on a [third-party macOS VNC client](https://edovia.com/screens-mac) that actually works, even if the developers kind of disdain Linux VNC servers.

**Reply by elsiehupp** [reply]
That said, it would be super nice if someone who knows their way around `deb` packaging were willing to put in the work to submit `wayvnc` to one or more official `apt` repositories. I just don’t think I’m the person for the job, unfortunately.

**Reply by okc** [reply]
You can compile it under ubuntu after installing requirements and prerequisites, but if you try to run it you get the same problem as issue #100 and #104 which were summarily closed, i.e.:

> `$ ./wayvnc
>   wl_registry@2: error 0: invalid version for global wl_seat (17): have 5, wanted 7
>   ERROR: Virtual Pointer protocol not supported by compositor.
>   ERROR: Failed to initialise wayland`

These error messages are not very helpful:
> Your wayland compositor (Gnome) does not support the required protocols. Wayvnc works for wlroots based compositors such as sway, hikari and wayfire.


If I move the executable to somewhere on the standard path, it can't find the shared object file.  Is that a setting in the config file?
> `$ wayvnc
>    wayvnc: error while loading shared libraries: libaml.so.0: cannot open shared object file: No such file or directory`
  
Then you create a systemctl service (cribbed from some other vnc server), package it up using the requirements knowledge from the meson build / ninja build files.

Not much point in packaging this project for ubuntu without support gnome or kde, as those are probably 99% of the ubuntu desktops.  What is involved in switching from gnome to sway or hikari or wayfire, and why would I want to do that?  Or can gnome's protocols be added?

**Reply by any1** [reply]
I wouldn't be opposed to a PR implementing Gnome support, but I'd much rather that the Gnome people implement proper wayland protocols for screen capturing instead of putting everything on dbus.

**Comment by crazy-logic** [comment]
For what it's worth - a lot of the X based VNC servers also have issues with the lock and login screens - I've only ever scratched the surface as to why this is the case but it's something along the lines of the Xsession auth, and that changes for the login/lock screen but the VNC server doesn't. I'm unsure, but could the same be happening with Wayland?
